### DISCUSSION TITLE
Caching while opening the page

### DISCUSSION THREAD

**Discussion by ankitk94** [dailydotdev/daily#798] [Feature Requests]
## Feature request
Implement time based caching when the data is loaded (when new tab is opened).

## Issue

Hi team, when I open a new tab I expect it to load almost instantly. But when I open a new tab now, it shows loading and takes some time to come up. Sometime I just need to open a site and always it fetches data from server.

#### Following API call is made:
https://api.daily.dev/graphql

#### With this request data as query:
"\n query Feed(\n $loggedIn: Boolean! = false\n $first: Int\n $after: String\n $ranking: Ranking\n $unreadOnly: Boolean\n $version: Int\n ) {\n page: feed(\n first: $first\n after: $after\n ranking: $ranking\n unreadOnly: $unreadOnly\n version: $version\n ) {\n ...FeedPostConnection\n }\n }\n \n fragment FeedPostConnection on PostConnection {\n pageInfo {\n hasNextPage\n endCursor\n }\n edges {\n node {\n ...FeedPost\n ...UserPost @include(if: $loggedIn)\n }\n }\n }\n \n fragment FeedPost on Post {\n id\n title\n createdAt\n image\n readTime\n source {\n id\n name\n image\n }\n permalink\n numComments\n numUpvotes\n commentsPermalink\n scout {\n id\n name\n image\n username\n }\n author {\n id\n name\n image\n username\n permalink\n }\n trending\n tags\n }\n\n \n fragment UserPost on Post {\n read\n upvoted\n commented\n bookmarked\n }\n\n\n"

It takes around 1.13 seconds to complete even on a very fast connection. Moreover if I am opening new tab continuously like 5-510 times, every time the call is made even if the content is same.

## Impact
Zero loading time once the data was loaded in let's say last 1 hour.

## Open for contribution
YES

**Comment by idoshamun** [comment]
Hi @ankitk94,

Thanks for the suggestion. I understand your concern, let me log that, and we'll think of ways to improve the loading time
